lib/widgets/message_list.dart

```diff
@@ -247,6 +244,30 @@ class _MessageListState extends State<MessageList> with PerAccountStoreAwareStat
   Widget _buildListView(context) {
     final length = model!.items.length;
     return StickyHeaderListView.builder(
+      findChildIndexCallback: (Key key) {
+        // To preserve state across rebuilds for individual [MessageItem]
+        // widgets as the size of [MessageListView.items] changes we need
+        // to match old widgets by their key to their new position in
+        // the list.
+        //
+        // The keys are of type [ValueKey] with a value of [Message.id]
+        // and here we use a O(log n) binary search method. This could
+        // be improved but for now it only triggers for materialized
+        // widgets. As a simple test, flinging through All Messages in
+        // CZO on a Pixel 5, this only runs about 10 times per rebuild
+        // and the timing for each call is <100 microseconds.
+        //
+        // Non-message items (e.g., start and end markers) that do not
+        // have state that needs to be preserved have not been given keys
+        // and will not trigger this callback.
+        final valueKey = key as ValueKey;
+        final index = model!.findItemWithMessageId(valueKey.value);
+        if (index == -1) {
+          return null;
+        }
+        // Note `itemBuilder` pulls from `model.items` in reverse order.
+        return length - 1 - index;
+      },
       // TODO: Offer `ScrollViewKeyboardDismissBehavior.interactive` (or
```

Comment:
nit:
```suggestion
      },

      // TODO: Offer `ScrollViewKeyboardDismissBehavior.interactive` (or
```

Each of these is a sizable chunk of code, and they're fairly separate from each other in meaning, so the blank line helps visually set them off.

Especially since the latter chunk starts with a comment — the blank line separates it from the unrelated chunk above, helping it stay visually attached to the chunk below.